src/plonk_util.rs

```diff
@@ -257,6 +246,36 @@ pub(crate) fn polynomial_degree<F: Field>(
     coeffs.iter().rev().skip_while(|c| c.is_zero()).count() - 1
 }
 
+pub fn build_s<F: Field>(us: &[F]) -> Vec<F> {
```

Comment:
Could you either call it `halo_s`, or add a comment to mention that it's `s` in the context of the Halo paper?

Reply:
I changed the name and remove the `pub` since I only use it to test the `g` function.